dig.go

```diff
@@ -641,16 +664,13 @@ func (c *Container) provide(ctor interface{}, opts provideOptions) error {
 	}
 
 	for k := range keys {
-		c.isVerifiedAcyclic = false
-		oldProviders := c.providers[k]
 		c.providers[k] = append(c.providers[k], n)
```

Comment:
actually this might be an unintentional behavior change because c.providers[k] should not be changed when a cycle is introduced by a newly provided constructor, but with the current changes, it will be. Let me fix that.